presto-main/src/main/java/com/facebook/presto/operator/unnest/ArrayUnnester.java

```diff
@@ -65,22 +58,22 @@ protected Block getElementsBlock(int channel)
     }
 
     @Override
-    protected void processCurrentPosition(int requiredOutputCount)
+    protected int getElementsLength(int index)
     {
-        // Translate indices
-        int startElementIndex = columnarArray.getOffset(getCurrentPosition());
-        int length = columnarArray.getLength(getCurrentPosition());
-
-        // Append elements and nulls
-        getBlockBuilder(0).appendRange(startElementIndex, length);
-        for (int i = 0; i < requiredOutputCount - length; i++) {
-            getBlockBuilder(0).appendNull();
-        }
+        return columnarArray.getLength(index);
     }
 
-    @Override
-    protected int getElementsLength(int index)
+    protected boolean needToInsertNulls(int offset, int length, int maxTotalEntriesForBatch, int unnesterCount)
     {
-        return columnarArray.getLength(index);
+        if (unnesterCount == 1) {
+            return false;
+        }
+
+        int totalEntriesForBatch = columnarArray.getOffset(offset + length) - columnarArray.getOffset(offset);
+        if (totalEntriesForBatch < maxTotalEntriesForBatch) {
```

Comment:
- totalEntriesForBatch -> totalEntries for readability
- replace `if` with `return`
- ditto in MapUnnester

```
        int totalEntries = columnarArray.getOffset(offset + length) - columnarArray.getOffset(offset);
        return totalEntries < maxTotalEntriesForBatch;
```